lib/__tests__/cli-suppressions.test.mjs

```diff
@@ -0,0 +1,554 @@
+			);
+		});
+
+		it('creates the suppressions file when --suppress-location specifies a directory', async () => {
+			const supDir = path.join(cssTmpDir, 'suppressions-dir');
+			const expectedSupPath = path.join(supDir, 'stylelint-suppressions.json');
+
+			await fs.mkdir(supDir, { recursive: true });
+
+			await cli([
+				'--suppress',
+				'--suppress-location',
+				supDir,
+				'--config',
+				CONFIG_BLOCK_ONLY,
+				EMPTY_BLOCK_CSS,
+			]);
+
+			expect(process.exitCode).toBeUndefined();
+
+			expect(await fs.readFile(expectedSupPath, 'utf8')).toMatch(/"block-no-empty"/);
+
+			expect(process.stderr.write).toHaveBeenCalledWith(
+				expect.stringMatching(/The suppressions feature is experimental/),
+				expect.any(Function),
+			);
+		});
+
+		it('creates the suppressions file when the --suppress=<rule> flag is used, and reports some violations', async () => {
+			await cli([
+				'--suppress=color-no-invalid-hex',
+				'--suppress-location',
+				SUPPRESSIONS_PATH,
+				'--config',
+				CONFIG_BLOCK_AND_COLOR,
+				EMPTY_BLOCK_CSS,
+				INVALID_HEX_CSS,
+			]);
+
+			expect(process.exitCode).toBe(EXIT_CODE_LINT_PROBLEM);
+
+			const sup = JSON.parse(await fs.readFile(SUPPRESSIONS_PATH, 'utf8'));
+
+			const relativeInvalidHex = getRelativePath(process.cwd(), INVALID_HEX_CSS);
+
+			expect(sup[relativeInvalidHex]).toHaveProperty('color-no-invalid-hex');
+
+			const relativeEmptyBlock = getRelativePath(process.cwd(), EMPTY_BLOCK_CSS);
+
+			expect(sup[relativeEmptyBlock]).toBeUndefined();
+
+			expect(process.stderr.write).toHaveBeenCalledTimes(2);
+			expect(process.stderr.write.mock.calls[0][0]).toMatch(
+				/The suppressions feature is experimental/,
+			);
+			expect(process.stderr.write.mock.calls[1][0]).toMatch(/block-no-empty/);
+			expect(process.stderr.write.mock.calls[1][0]).not.toMatch(/color-no-invalid-hex/);
+		});
+
+		it('creates the suppressions file when multiple --suppress=<rule> flags are used, and reports some violations', async () => {
+			await cli([
+				'--suppress=color-no-invalid-hex',
+				'--suppress=block-no-empty',
+				'--suppress-location',
+				SUPPRESSIONS_PATH,
+				'--config',
+				CONFIG_BLOCK_COLOR_LENGTH,
+				EMPTY_BLOCK_CSS,
+				INVALID_HEX_CSS,
+				LENGTH_ZERO_NO_UNIT_CSS,
+			]);
+
+			expect(process.exitCode).toBe(EXIT_CODE_LINT_PROBLEM);
+
+			const sup = JSON.parse(await fs.readFile(SUPPRESSIONS_PATH, 'utf8'));
+
+			expect(Object.keys(sup)).toHaveLength(2);
+
+			expect(process.stderr.write).toHaveBeenCalledTimes(2);
+			expect(process.stderr.write.mock.calls[0][0]).toMatch(
+				/The suppressions feature is experimental/,
+			);
+			expect(process.stderr.write.mock.calls[1][0]).toMatch(/length-zero-no-unit/);
+			expect(process.stderr.write.mock.calls[1][0]).not.toMatch(/block-no-empty/);
+			expect(process.stderr.write.mock.calls[1][0]).not.toMatch(/color-no-invalid-hex/);
+		});
+
+		it("displays an error when the suppressions file doesn't exist", async () => {
```

Comment:
question

Do we allow the usage of `--suppress-location` without `--suppress`?
i.e. does `--suppress-location` implies `--suppress`?

ditto l238